### PR TITLE
Show error message on `capture` fail

### PR DESCRIPTION
**Pull request by dswij**
Handle and print more readable error messages on `capture` fail rather than panicking.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -35,6 +35,12 @@ jobs:
           command: build
           args: --release --all-features
 
+      - name: Rustfmt
```
Comment (tkellogg) [inline .github/workflows/build.yaml:38]: awww this is amazing!

File: src/main.rs [hunk 2]
```diff
@@ -85,8 +86,16 @@ async fn main() {
...
-            if let Some(oid) = snapshots::capture(dir).unwrap() {
-                println!("{}", oid);
+            match snapshots::capture(dir) {
+                Ok(oid_opt) => {
+                    if let Some(oid) = oid_opt {
+                        println!("{}", oid);
+                    }
+                }
+                Err(e) => {
+                    println!("Dura capture failed: {}", e);
```
Comment (tkellogg) [inline src/main.rs:96]: 👍

File: tests/util/daemon.rs [hunk 3]
```diff
@@ -10,19 +10,22 @@ use std::time::Duration;
 /// tests more reliable when they dispatch asynchronously to `dura serve`. However, nothing abot
 /// this is intended to be specific to dura.
 pub struct Daemon {
-   mailbox: Receiver<Option<String>>, 
-   pub child: Child,
-   /// Signals to kill daemon thread if this goes <= 0, like a CountDownLatch
-   kill_sign: Arc<Mutex<i32>>,
+    mailbox: Receiver<Option<String>>,
```
Comment (tkellogg) [inline tests/util/daemon.rs:13]: i'm not sure why my editor started doing 3 space tabs. i'm embarrassed, actually. i saw it before but couldn't make sense of what was going on

**Review by tkellogg — APPROVED** [review]

File: tests/util/daemon.rs (hunk 3 above)
Comment (dswij) [reply tests/util/daemon.rs:13]: FWIW, I've betrayed my editor because of Rust 😄

**Pull request merged**